examples/demo-app/src/cloud-providers/aws/aws-provider.js

```diff
@@ -0,0 +1,291 @@
+// Copyright (c) 2020 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+import AwsIcon from './aws-icon';
+import {Provider} from 'kepler.gl/cloud-providers';
+import window from 'global';
+import 'aws-sdk';
+import {Auth, Storage} from 'aws-amplify';
+import {MAP_URI} from '../../constants/default-settings';
+import {AWS_LOGIN_URL, AWS_WEB_CLIENT_ID} from './aws-login';
+
+export const PROVIDER_NAME = 'aws';
+export const DISPLAY_NAME = 'AWS';
+export const PRIVATE_STORAGE_ENABLED = true;
+export const SHARING_ENABLED = true;
+
+Storage.configure({level: 'private'});
+
+export default class AwsProvider extends Provider {
+  constructor() {
+    super({name: PROVIDER_NAME, displayName: DISPLAY_NAME, icon: AwsIcon});
+  }
+
+  /**
+   * Required!
+   * Idea / Hack: https://github.com/amazon-archives/amazon-cognito-identity-js/issues/508
+   *
+   * @param onCloudLoginSuccess
+   */
+  async login(onCloudLoginSuccess) {
+    const link = `${window.location.protocol}//${window.location.host}/${AWS_LOGIN_URL}`;
+    const style = `location, toolbar, resizable, scrollbars, status, width=500, height=440, top=200, left=400`;
+    const authWindow = window.open(link, 'awsCognito', style);
+
+    const handleLogin = async (e) => {
+      if (authWindow) {
+        authWindow.close();
+      }
+
+      window.removeEventListener('message', handleLogin);
+
+      if (e.data.success) {
+        onCloudLoginSuccess();
+      }
+    };
+    window.addEventListener('message', handleLogin);
```

Comment:
The `message` is listener is quite generic. How do you know this message was meant for your provider?

Reply:
My understanding is that the listener is listening on the provided link only. The other providers typically have a different auth route.

Reply:
@stgrmks are you sure about this? I am not so sure here.

Reply:
Thomas, if we send the provider name and check on it, you think, this would be enough to ensure we are listening to the right event? 

**aws-login.js:**
        window.opener.postMessage({success: true, provider: 'aws'}, location.origin);

**aws-provider.js**
  const handleLogin = async e => {
    if (e.data && e.data.provider && e.data.provider !== 'aws'){
            return
       }
  ....
}

Reply:
@Pekxy We can guarantee that messages from our URI are intended for our provider, since we created the routing for it in main.js.
But I am not sure if this is the issue rollacaster has in mind.

Reply:
It's the issue I had in mind. I just noticed there's also a comment about that in the `dropbox-provider`. I think it could happen that you're trying to access dropbox & aws and you since both are listening on `message` both receive all messages

Reply:
(If you're listener was not removed yet due to a successful login)